tinygrad/ops.py

```diff
@@ -7,6 +7,8 @@
 def unbroadcast(out, in_sh):
   # adjoint operation to broadcast is sum. Need to sum all axis with 1 = in_sh[i] < out.shape[i]
   sum_axis = [i for i in range(len(in_sh)) if in_sh[i]==1 and out.shape[i]>1]
+  if not in_sh: # incase input is a scalar.
```

Comment:
a hackish way... i know ..
im not sure if its readable 
@geohot can u suggest something here?

Reply:
Oh, this shouldn't be in ops

Reply:
well @geohot,
if input is scalar i.e 2.0 * Tensor([1.0,2.0])
this 2.0 scalar is converted to a numpy array with shape (),
but during multiplication in 

> Looking good!
> 
> Absolutely yes on right operations.
> 
> Though we shouldn't be playing code golf. If you want more lines, refactor something else. If it's not the cleanest way to write it, don't. I will merge even if it breaks the line limit if it's clean, then I'll save the lines myself.

point noted :) @geohot

Reply:
@geohot for scalar multiplications i.e 
Tensor object with multidimensional data * 2.0
this 2.0 is converted to numpy.array with shape = ()
but due to broadcasting this numpy changes to dimension of Tensor for multiplication

and for backward pass  inside broadcasting y.shape is still = (), and out_grad is mulltidimensional hence  out_grad wont be compatible with y as it just should be scalar. so i did , out.flat[0] which basically picks first element. :)

Reply:
Convert it to a numpy array with shape (1,) instead.

Reply:
cannot reshape (2925,) to (1,)

Reply:
@geohot got it , just need to change sum_axis = None :)